### PR TITLE
Upgrade flyteidl, preload lb policy and add default service config for catalog client

### PR DESCRIPTION
**Pull request by honnix**
# TL;DR
Upgrade flyteidl and preload lb policy.

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [x] Unit tests added
 - [x] Code documentation added
 - [ ] Any pending items have an associated Issue

## Complete description
This is to take in https://github.com/flyteorg/flyteidl/pull/301 and preload `roundrobin` load balancing policy.

This PR also adds the capability to config catalog gRPC client with default service config.

## Tracking Issue

fixes flyteorg/flyte#2510

## Follow-up issue
_NA_

### CODE REVIEW

**Review by EngHabu — DISMISSED** [review]

**Comment by honnix** [comment]
@EngHabu Thanks for approving this. I also worked on https://github.com/honnix/flytepropeller/pull/1. Do you prefer we merge that into this PR, or ship this and then rebase? Thanks.

**Comment by EngHabu** [comment]
@honnix I agree... let's merge that into this first...

**Comment by honnix** [comment]
> @honnix I agree... let's merge that into this first...

Done.

File: pkg/controller/nodes/task/catalog/config_flags.go [hunk 1]
```diff
@@ -55,5 +55,6 @@ func (cfg Config) GetPFlagSet(prefix string) *pflag.FlagSet {
 	cmdFlags.Bool(fmt.Sprintf("%v%v", prefix, "insecure"), defaultConfig.Insecure, " Use insecure grpc connection")
 	cmdFlags.String(fmt.Sprintf("%v%v", prefix, "max-cache-age"), defaultConfig.MaxCacheAge.String(), " Cache entries past this age will incur cache miss. 0 means cache never expires")
 	cmdFlags.Bool(fmt.Sprintf("%v%v", prefix, "use-admin-auth"), defaultConfig.UseAdminAuth, " Use the same gRPC credentials option as the flyteadmin client")
+	cmdFlags.String(fmt.Sprintf("%v%v", prefix, "default-service-config"), defaultConfig.DefaultServiceConfig, " The default service config applied to gRPC client")
```
Comment (honnix) [inline pkg/controller/nodes/task/catalog/config_flags.go:58]: Looks like I modified a generated file. Where is it generated from?

Comment (honnix) [reply pkg/controller/nodes/task/catalog/config_flags.go:58]: 🙏

File: propeller-config.yaml [hunk 2]
```diff
@@ -99,10 +99,12 @@ event:
 admin:
   endpoint: localhost:30081
   insecure: true
+  defaultServiceConfig: '{"loadBalancingConfig": [{"round_robin":{}}]}'
 catalog-cache:
   type: noop
   endpoint: datacatalog:8089
   insecure: true
+  default-service-config: '{"loadBalancingConfig": [{"round_robin":{}}]}'
```
Comment (honnix) [inline propeller-config.yaml:107]: This inconsistency is annoying but I guess hard to fix now.

**Review by EngHabu — APPROVED** [review]

**Pull request merged**